### PR TITLE
Added ability to collect Connected Devices for flow

### LINKED ISSUE #2582: Add ability to get a list of devices connected to Flow srt/dst endpoint
There are 2 ways to get a list of connected devices: LLDP packets and ARP packets.
Both of them will be used.

Customer's traffic may contain LLDP packets. Kilda must catch such packets to send one copy of a packet to controller and the other copy to Flow endpoint.

API changes:
Following APIs will be extended by `show_connected_devices` field:
`GET /flow`
`PUT /flow`
`GET /flow/{flow_id}`
`PUT /flow/{flow_id}`
`PATCH /flow/{flow_id}`

new Request body will be looks like:

```{
  "flowid": "string",
  "source": {
    "port-id": 0,
    "switch-id": "string",
    "vlan-id": 0
    "show_connected_devices": true
  },
  "destination": {
    "port-id": 0,
    "switch-id": "string",
    "vlan-id": 0,
    "show_connected_devices": true
  },
  "maximum-bandwidth": 0,
  ****
```

With new field `show_connected_devices` user can enable/disable collecting of connection devices for flow source and destination separately 

To get list of connected devices new API will be introduced:
`GET flows/{flow-id}/devices`

Response:
```
{
   "source": [
     { "chassis_id": "string",
        "port_id": "string",
        "ttl": int,
        "system_name": "string",
        "system_description": "string",
        "system_capabilities": "string",
        "management_address": "string"
     },
     {
          ****
     }],
   "destination": [
      { "chassis_id": "string",
        "port_id": "string",
        "ttl": int,
        "system_name": "string",
        "system_description": "string",
        "system_capabilities": "string",
        "management_address": "string"
      },
      {
           ****
      }]
}
```

### PR DESCRIPTION
**Pull request by niksv**
We need to add the indexes to the database:
```
CREATE INDEX ON :`connected_device` (`chassis_id`);
CREATE INDEX ON :`connected_device` (`port_id`);
```

Closes #2582

### CODE REVIEW

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/FlowService.java [hunk 1]
```diff
@@ -205,4 +206,13 @@
      * @return the list of updated flows.
      */
     CompletableFuture<SwapFlowEndpointPayload> swapFlowEndpoint(SwapFlowEndpointPayload input);
+
+    /**
+     * Get Flow connected devices.
+     *
+     * @param flowId flow ID
+     * @param since since
```
Comment (dpoltavets) [inline services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/FlowService.java]: What does this parameter mean?

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/FlowController.java [hunk 2]
```diff
@@ -358,4 +359,20 @@ public void invalidateFlowCache() {
...
+    @GetMapping(path = "/{flow_id}/devices")
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<FlowConnectedDevicesResponse> getHistory(
+            @PathVariable("flow_id") String flowId,
+            @ApiParam(value = "Device will be included in response if it's `time_last_seen` >= `since`. "
+                    + "Example of `since` value: `2019-09-30T16:14:12.538Z`",
+                    required = false)
+            @RequestParam(value = "since", required = false) Optional<String> since) {
+
+        return flowService.getFlowConnectedDevices(flowId, since.orElse(""));
```
Comment (sergii-iakovenko) [inline services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/FlowController.java]: Why not parse the date-time field right in the controller?

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/FlowService.java (hunk 1 above)
Comment (niksv) [reply services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/FlowService.java]: fixed

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/FlowController.java (hunk 2 above)
Comment (niksv) [reply services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/FlowController.java]: fixed

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/ConnectedDeviceMapper.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.openkilda.northbound.dto.v1.flows.TypedConnectedDevicesDto;
+
+import org.mapstruct.Mapper;
+
+@Mapper(componentModel = "spring")
+public interface ConnectedDeviceMapper {
+
+    ConnectedDeviceDto map(org.openkilda.messaging.nbtopology.response.ConnectedDeviceDto deviceDto);
+
+    org.openkilda.messaging.nbtopology.response.ConnectedDeviceDto map(ConnectedDeviceDto deviceDto);
```
[the remark this answers is not in the document]
Comment (niksv) [reply services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/ConnectedDeviceMapper.java]: useless methods were removed

**Pull request merged**